dspace-api/src/main/java/org/dspace/iiif/canvasdimension/IIIFCanvasDimensionServiceImpl.java

```diff
@@ -198,6 +199,7 @@ private boolean processBitstream(Context context, Bitstream bitstream) throws SQ
                 }
             }
         }
+        context.uncacheEntity(bitstream);
```

Comment:
This location seems OK.  But, I'd again point out that the "best" place to uncache tends to be within a larger `for` or `while` loop.  So, technically, it's slightly more accurate to move this into the `for` loop inside `processItemBundles()`.  But, based on the current code, the behavior would currently be the same as if it was in this location.

Reply:
I agree. I'll move to the for loop.

Reply:
I stepped back and looked at `uncacheEntity()`. On face value, it looks like uncaching the individual bitstreams as they are processed is not all that helpful. But it actually does improve performance.

My observed runtimes for a small collection of 99 documents of varying size (some documents rather large):

- Only `uncacheEntity(item)`: ~155 seconds
- Only `uncacheEntity(bitstream)`: ~300 seconds
- Both `uncacheEntity(item)` and `uncacheEntity(bitstream)`: ~86 seconds
- Neither: ~`infinity`